### PR TITLE
PDCL-9504 - added functional tests for client hints in the Personalization module

### PR DESCRIPTION
**Pull request by dcottingham**
## Description
https://jira.corp.adobe.com/browse/PDCL-9504
Added two new functional tests to verify client hints based targeting in the Personalization module.  One tests low entropy hints; the other tests high entropy hints.  

## Motivation and Context

These tests verify Target-specific functionality of client hints when used as part of Target audiences/profile scripts.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

**Review by jonsnyder — APPROVED** [review]
Nice tests!

File: test/functional/specs/Personalization/C8631576.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+test(DESCRIPTION, async () => {
+  const alloy = createAlloyProxy();
+  await alloy.configure(config);
+  const eventResult = await alloy.sendEvent(sendEventOptions);
+
+  await responseStatus(networkLogger.edgeEndpointLogs.requests, 200);
+  await t.expect(networkLogger.edgeEndpointLogs.requests.length).eql(1);
+
+  // Tests must be run using https otherwise this will return false
+  if (await isUserAgentClientHintsSupported()) {
```
Comment (carterworks) [inline test/functional/specs/Personalization/C8631576.js:43]: I'm always a little wary of conditionals in tests, so I just want to confirm the behavior: if an end user client device does not support user agent client hints, then they should not qualify for the experiences, and this test is also testing that outcome. "
Does that sound correct?

**Review by carterworks — APPROVED** [review]

**Review by carterworks — APPROVED** [review]

File: test/functional/specs/Personalization/C8631576.js (hunk 1 above)
Comment (dcottingham) [reply test/functional/specs/Personalization/C8631576.js:43]: @carterworks, that's a great point - I will add some assertions for the case when client hints are not supported (i.e. user is on Firefox or Safari).  You're correct that in that case the user should not qualify for the experience.

Comment (carterworks) [reply test/functional/specs/Personalization/C8631576.js:46]: Nicely done.

**Pull request merged**